workflow/asynchronous-in-js/callback.js

```diff
@@ -1,107 +1,95 @@
-function loadOneByOneCallback() {
-  function loadImage(url, callback, whatToCall) {
+const loadOneByOneCallback = () => {
+  const loadImage = (url, whatToCall) => {
     createAndAppendSpinnerDiv(createNewId());
 
     const xhr = new XMLHttpRequest();
     xhr.open('GET', url);
-    xhr.onload = function () {
-      const src = JSON.parse(xhr.response)[0].url;
-      callback(null, src, whatToCall);
+    xhr.onload = () => {
+      const src = JSON.parse(xhr.response || null)?.[0]?.url;
+      createImage(null, src, whatToCall);
     };
-    xhr.onerror = function () {
-      callback(error);
+    xhr.onerror = () => {
+      createImage(error);
     };
     xhr.send();
-  }
+  };
 
-  function createImage(error, src, collLoadImage) {
+  const createImage = (error, src, cb) => {
     if (error) {
       alert(error);
     } else {
       createAndAppendImage(src, startId);
-      collLoadImage?.();
+      cb && cb();
     }
-  }
+  };
 
-  loadImage(url, createImage, () =>
-    loadImage(url, createImage, () =>
-      loadImage(url, createImage, () => loadImage(url, createImage, () => loadImage(url, createImage))),
-    ),
-  );
-}
+  loadImage(url, () => loadImage(url, () => loadImage(url, () => loadImage(url, () => loadImage(url)))));
```

Comment:
it's still callback hell, but it's okay, the way I recommended (with load(url)(url)(url)) suites for simultaneous requests